internal/controller/dnsrecord_controller.go

```diff
@@ -123,17 +123,26 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	setDNSRecordCondition(dnsRecord, string(conditions.ConditionTypeReady), status, reason, message)
 	dnsRecord.Status.ObservedGeneration = dnsRecord.Generation
 	dnsRecord.Status.Endpoints = dnsRecord.Spec.Endpoints
+
+	if err := r.ReconcileHealthChecks(ctx, dnsRecord); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return r.updateStatus(ctx, previous, dnsRecord)
 }
 
 func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord) (reconcile.Result, error) {
 	if !equality.Semantic.DeepEqual(previous.Status, current.Status) {
+		log.Log.Info("status changed", "previous", previous.Status, "current", current.Status)
```

Comment:
might be a bit noisey?

Reply:
these were primarily for debugging, I will remove a bunch of them now.